src/segment-loader.js

```diff
@@ -355,6 +360,34 @@ export default class SegmentLoader extends videojs.EventTarget {
     return storedMap || map;
   }
 
+  /**
+   * Gets and sets key for the provided key
+   *
+   * @param {Object} key
+   *        The key object representing the key to get or set
+   * @param {Boolean=} set
+   *        If true, the key for the provided key should be saved
+   * @return {Object}
+   *         Key object for desired key
+   */
+   segmentKey(key, set = false) {
+     if (!key) {
+       return null;
+     }
+
+     const id = segmentKeyId(key);
+     let storedKey = this.keyCache_[id];
+
+     if (this.cacheEncryptionKeys_ && set && !storedKey && key.bytes) {
+       this.keyCache_[id] = storedKey = {
```

Comment:
we may also want to check if the key has actually changed? I suppose this would only be if the iv value is different

Reply:
A change in IV does not mean the key has changed. Probably the proper approach is to check for the HTTP Expires header and invalidate our cache based on that per https://tools.ietf.org/html/draft-pantos-http-live-streaming-23#section-6.2.3